config/routes.rb

```diff
@@ -26,6 +26,9 @@
     end
   end
 
+  patch '/roleupdate', to: 'users#roleupdate'
+  put '/roleupdate', to: 'users#roleupdate'
```

Comment:
Why do you have a put?

Reply:
When I only had patch and tried to run it, it crashed and said it also needed put.

Reply:
aight aight, then maybe change it to resources? Or change the resources to be like this? You can make a new task like refactor routes.rb :) 👍